client/client_test.go

```diff
@@ -3617,38 +3617,45 @@ func TestGetAllTargetInfo(t *testing.T) {
 	// level2 - signed by targets/level2 and targets/level1/level2, with the same hash
 
 	// Positive cases
-	targetData, err := repo.GetAllTargetMetadataByName("current")
+	targetSignatureData, err := repo.GetAllTargetMetadataByName("current")
 	require.NoError(t, err)
-	require.NotNil(t, targetData)
-	require.Len(t, targetData, 3)
-	require.Equal(t, targetData[data.CanonicalTargetsRole], *targetsCurrentTarget)
-	require.Equal(t, targetData["targets/level1"], *level1CurrentTarget)
-	require.Equal(t, targetData["targets/level2"], *level2CurrentTarget)
+	require.NotNil(t, targetSignatureData)
+	require.Len(t, targetSignatureData, 3)
+	require.Len(t, targetSignatureData[data.CanonicalTargetsRole].Signatures, 1)
+	require.Equal(t, (targetSignatureData[data.CanonicalTargetsRole]).Target, *targetsCurrentTarget)
+	require.Equal(t, (targetSignatureData["targets/level1"]).Target, *level1CurrentTarget)
+	require.Len(t, targetSignatureData["targets/level1"].Signatures, 1)
+	require.Equal(t, (targetSignatureData["targets/level2"]).Target, *level2CurrentTarget)
+	require.Len(t, targetSignatureData["targets/level2"].Signatures, 1)
```

Comment:
In addition to checking the length of the signature, can we check that the signature `KeyID` corresponds to the role's key, and that the signature verifies correctly?